src/LogicTasks/Semantics/Fill.hs

```diff
@@ -146,21 +147,16 @@ partialGrade FillInst{..} sol = do
       german $ "Die Lösung muss genau "  ++ show missingLen ++ " Lücken enthalten."
       english $ "The solution must contain exactly " ++ show missingLen ++ " gaps."
     )
-  pure ()
-  where
-    tableLen = length $ readEntries $ getTable cnf
-    missingLen = length missing
-    solLen = length sol
-    diffToTable = abs (solLen - tableLen)
-    diffToMissing = abs (solLen - missingLen)
-    (gerLong,engLong) = gerEng diffToTable
-    (gerShort,engShort) = gerEng diffToMissing
-    gerEng diff = if diff == 1
-        then ("muss " ++ display ++ " Wert", display ++ " value") -- no-spell-check
-        else ("müssen " ++ display ++ " Werte", display ++ " values") -- no-spell-check
-      where
-        display = show diff
 
+  preventIfSolutionExceedsTableSize (length sol) (getTable cnf)
```

Comment:
Is it even possible that execution reaches this point and `length sol` is greater than the number of table rows? If that were the case, then already `not (solLen == tableLen || solLen == missingLen)` from above would have been fulfilled, and hence the input be rejected outright, it seems.